src/onnx/parse_reduce_op.cpp

```diff
@@ -15,15 +16,42 @@ instruction_ref parse_reduce_oper(const std::string& op_name,
     std::size_t n_dim = args.front()->get_shape().lens().size();
 
     // default to reduce over all dimensions
-    std::vector<int64_t> axes(n_dim);
-    std::iota(axes.begin(), axes.end(), 0);
-    if(contains(info.attributes, "axes"))
+    std::vector<int64_t> axes;
+    if(args.size() == 2)
+    {
+        auto arg_axes = args.at(1)->eval();
+        check_arg_empty(arg_axes, "PARSE_" + op_name + ": cannot handle variable axes!");
+        axes.clear();
+        arg_axes.visit([&](auto s) { axes.assign(s.begin(), s.end()); });
+    }
+    else if(contains(info.attributes, "axes"))
     {
         axes.clear();
         auto&& attr_axes = info.attributes["axes"].ints();
         axes             = std::vector<int64_t>(attr_axes.begin(), attr_axes.end());
     }
 
+    int noop_with_empty_axes = 0;
```

Comment:
Shouldn't this be a `bool`?

Reply:
The ONNX operator uses an int type, but I can change it to bool here.